MotionPhotoReader/app/src/main/java/com/example/motionphotoreader/XmpParser.java

```diff
@@ -0,0 +1,82 @@
+package com.example.motionphotoreader;
+
+import android.util.Log;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+import com.adobe.internal.xmp.XMPMetaFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+
+/**
+ * The XmpParser class is a package-private class intended to help extract the microvideo offset
+ * information from the XMP metadata of the given Motion Photo. This class is for internal use only.
+ */
+class XmpParser {
+
+    private static final byte[] OPEN_ARR = "<x:xmpmeta".getBytes();  /* Start of XMP metadata tag */
+    private static final byte[] CLOSE_ARR = "</x:xmpmeta>".getBytes();  /* End of XMP metadata tag */
+
+    /**
+     * Copies the input stream from a file to an output stream.
+     */
+    private static void copy(String filename, InputStream in, OutputStream out) throws IOException {
+        int len;
+        byte[] buf = new byte[1024];
+        while((len = in.read(buf)) >= 0) {
+            out.write(buf, 0, len);
+        }
```

Comment:
There's a really nice Guava method called ByteStreams.copy that you should consider using ([reference link](https://guava.dev/releases/19.0/api/docs/com/google/common/io/ByteStreams.html#copy(java.io.InputStream,%20java.io.OutputStream)))

Reply:
I used this method instead of the copy() one I wrote. There is a warning that it's marked unstable with @Beta but I haven't run into any issues yet.